\hypertarget{classpthread_1_1util_1_1queue__exception}{}\section{pthread\+:\+:util\+:\+:queue\+\_\+exception Class Reference}
\label{classpthread_1_1util_1_1queue__exception}\index{pthread\+::util\+::queue\+\_\+exception@{pthread\+::util\+::queue\+\_\+exception}}


{\ttfamily \#include $<$exceptions.\+hpp$>$}

Inheritance diagram for pthread\+:\+:util\+:\+:queue\+\_\+exception\+:\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[height=3.000000cm]{classpthread_1_1util_1_1queue__exception}
\end{center}
\end{figure}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\hyperlink{classpthread_1_1util_1_1queue__exception_abdae10e09e6f7cb24f64f0aee107590c}{queue\+\_\+exception} (const std\+::string \&msg=\char`\"{}queue\+\_\+error occured.\char`\"{})
\item 
virtual const char $\ast$ \hyperlink{classpthread_1_1util_1_1queue__exception_a41cf42358ff05d8df966713d3f2c64b3}{what} () const  noexcept override
\end{DoxyCompactItemize}
\subsection*{Protected Attributes}
\begin{DoxyCompactItemize}
\item 
std\+::string \hyperlink{classpthread_1_1util_1_1queue__exception_a16e98607fa8cdcbd76e16f8965db533e}{\+\_\+message}\hypertarget{classpthread_1_1util_1_1queue__exception_a16e98607fa8cdcbd76e16f8965db533e}{}\label{classpthread_1_1util_1_1queue__exception_a16e98607fa8cdcbd76e16f8965db533e}

\begin{DoxyCompactList}\small\item\em message buffer \end{DoxyCompactList}\end{DoxyCompactItemize}


\subsection{Detailed Description}
thrown when something goes wrong in a synchonized queue. 

Definition at line 126 of file exceptions.\+hpp.



\subsection{Constructor \& Destructor Documentation}
\index{pthread\+::util\+::queue\+\_\+exception@{pthread\+::util\+::queue\+\_\+exception}!queue\+\_\+exception@{queue\+\_\+exception}}
\index{queue\+\_\+exception@{queue\+\_\+exception}!pthread\+::util\+::queue\+\_\+exception@{pthread\+::util\+::queue\+\_\+exception}}
\subsubsection[{\texorpdfstring{queue\+\_\+exception(const std\+::string \&msg=""queue\+\_\+error occured."")}{queue\_exception(const std::string \&msg="queue\_error occured.")}}]{\setlength{\rightskip}{0pt plus 5cm}pthread\+::util\+::queue\+\_\+exception\+::queue\+\_\+exception (
\begin{DoxyParamCaption}
\item[{const std\+::string \&}]{msg = {\ttfamily \char`\"{}queue\+\_\+error~occured.\char`\"{}}}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [explicit]}}\hypertarget{classpthread_1_1util_1_1queue__exception_abdae10e09e6f7cb24f64f0aee107590c}{}\label{classpthread_1_1util_1_1queue__exception_abdae10e09e6f7cb24f64f0aee107590c}
new instance.


\begin{DoxyParams}{Parameters}
{\em msg} & explanatory message. \\
\hline
\end{DoxyParams}


Definition at line 64 of file exceptions.\+cpp.



\subsection{Member Function Documentation}
\index{pthread\+::util\+::queue\+\_\+exception@{pthread\+::util\+::queue\+\_\+exception}!what@{what}}
\index{what@{what}!pthread\+::util\+::queue\+\_\+exception@{pthread\+::util\+::queue\+\_\+exception}}
\subsubsection[{\texorpdfstring{what() const  noexcept override}{what() const  noexcept override}}]{\setlength{\rightskip}{0pt plus 5cm}const char $\ast$ pthread\+::util\+::queue\+\_\+exception\+::what (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
) const\hspace{0.3cm}{\ttfamily [override]}, {\ttfamily [virtual]}, {\ttfamily [noexcept]}}\hypertarget{classpthread_1_1util_1_1queue__exception_a41cf42358ff05d8df966713d3f2c64b3}{}\label{classpthread_1_1util_1_1queue__exception_a41cf42358ff05d8df966713d3f2c64b3}
\begin{DoxyReturn}{Returns}
an explanatory message 
\end{DoxyReturn}


Definition at line 70 of file exceptions.\+cpp.



The documentation for this class was generated from the following files\+:\begin{DoxyCompactItemize}
\item 
include/pthread/exceptions.\+hpp\item 
src/exceptions.\+cpp\end{DoxyCompactItemize}
